api/osb/store_instances_plugin.go

```diff
@@ -382,6 +382,13 @@ func (ssi *StoreServiceInstancePlugin) PollInstance(request *web.Request, next w
 		}
 
 		operationFromDB := op.(*types.Operation)
+		if response.StatusCode == http.StatusGone {
+			if operationFromDB.Type != types.DELETE {
+				return nil
```

Comment:
Shouldn't we proceed with the below logic or return here? I am not sure, looking at the code it seems to me that it should proceed

Reply:
A returned 410 Gone response will have no body, thus we cannot make the subsequent check to compare the state in the response with the last state in the DB. Also what would be the meaning of 410 Gone response to a CREATE/UPDATE operation?

And finally:

https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#response-1

"_Returning this while the Platform is polling for create or update operations SHOULD be interpreted as an invalid response and the Platform SHOULD continue polling._"